app/scripts/lib/fxa-client.js

```diff
@@ -98,14 +98,37 @@ function (_, FxaClient, $, xhr, p, Session, AuthErrors, Constants) {
     },
 
 
-    signIn: function (originalEmail, password, relier, options) {
+    /**
+     * Signin reasons are indicators to the backend that the signin
+     * is to complete a particular action. The backend can choose
+     * to perform actions, e.g., send emails, based on the reason.
+     */
+    SIGNIN_REASON: {
+      SIGN_IN: 'signin',
+      CHANGE_PASSWORD: 'change_password',
+      PASSWORD_RESET: 'password_reset',
```

Comment:
It's a trivial nit, but something about the inconsistent verb/noun ordering between "change password" and "password reset" just sits badly with me...how would you feel about "password_change" and "password_reset"?